llvm/unittests/Support/KnownBitsTest.cpp

```diff
@@ -281,7 +281,14 @@ TEST(KnownBitsTest, BinaryExhaustive) {
         return KnownBits::smin(Known1, Known2);
       },
       [](const APInt &N1, const APInt &N2) { return APIntOps::smin(N1, N2); });
-
+  testBinaryOpExhaustive(
+      [](const KnownBits &Known1, const KnownBits &Known2) {
+        return KnownBits::absdiff(Known1, Known2);
+      },
+      [](const APInt &N1, const APInt &N2) {
+        return APIntOps::absdiff(N1, N2);
+      },
+      checkCorrectnessOnlyBinary);
```

Comment:
Can you please also add some non-exhaustive tests to demonstrate that both of your absdiff implementations are necessary (i.e. if you remove either one, something should fail)?

Reply:
I've added an example for each absdiff path